common/libs/framing/frame_detector.h

```diff
@@ -0,0 +1,129 @@
+#ifndef __FRAME_DETECTOR__
+#define __FRAME_DETECTOR__
+
+#include "serial_listeners.h"
+#include <stdint.h>
+#include <string.h>
+
+template <class RxBuffer, int FRAME_BUF_LEN>
+class FrameDetector : public RxListener {
+public:
+  enum class State {
+    LOST,       // Frame detector does not know where it is in the stream of
+                // bytes coming into RxBuffer
+    WAIT_START, // Frame detector for a frame start marker
+    RX_FRAME    // Frame detector is receiving a frame
+  };
+
+  FrameDetector(RxBuffer &t) : rx_buffer_(t){};
+
+  // Starts frame detector
+  [[nodiscard]] bool Begin() {
+    state_ = State::LOST;
+    return rx_buffer_.Begin(this);
+  }
+
+  // Callback method called when RxBuffer is full
+  void onRxComplete() override {
+    // We should never reach the full read of rx buffer.
+    // If we get here, this means, there are no marker
+    // chars in the stream, so we are lost
+    error_counter_++;
+    state_ = State::LOST;
+    rx_buffer_.RestartRX(this);
+  }
+
+  // Callback method called when we receive a marker character
+  void onCharacterMatch() override {
+    switch (state_) {
+    case State::LOST:
+      // if we have received something before this marker,
+      // we assume, this is the frame end marker, so wait
+      // for start
+      if (rx_buffer_.ReceivedLength() > 1) {
+        state_ = State::WAIT_START;
+        // if we were lucky to get lost in the interframe silence,
```

Comment:
Is this comment referring to line 44 or to line 47? If to line 44, please merge with the comment above and condense; if to line 47, then please put before line 48.